cassandane/Cassandane/Cyrus/SearchFuzzy.pm

```diff
@@ -1453,6 +1453,45 @@ sub test_striphtml_alternative
     $self->assert_deep_equals([], $uids);
 }
 
+sub test_html_only
+    :min_version_3_3 :needs_search_xapian
+{
+    my ($self) = @_;
+    my $talk = $self->{store}->get_client();
+
+    xlog "Index message with both html and plain text part";
+    $self->make_message("test",
+        mime_type => "text/html",
+        body => ""
+          . "<html xmlns:o=\"urn:schemas-microsoft-com:office:office\">\r\n"
+          . "<div>This is an html <o:p>LL123</o:p> <h11>xyzzy</h11> body.</div>\r\n"
+          . "</html"
+
+    ) || die;
+    $self->{instance}->run_command({cyrus => 1}, 'squatter');
+
+    xlog "Assert that HTML in plain text is stripped";
+    my $uids = $talk->search('fuzzy', 'body', 'html') || die;
+    $self->assert_deep_equals([1], $uids);
+
+    $uids = $talk->search('fuzzy', 'body', 'div') || die;
+    $self->assert_deep_equals([], $uids);
+
+    # make sure the "p" doesn't leak into a token
+    $uids = $talk->search('fuzzy', 'body', 'LL123p') || die;
+    $self->assert_deep_equals([], $uids);
```

Comment:
Do we need a negative check for 'pLL123p' too?  Since it would have choked on the opening tag too...

I assume this check is based on what you saw happening in practice, that the dregs from the opening tag probably wind up as their own word, since there's a space before the opening tag, and it's only the dregs from the closing tag that get wrongly concatenated onto the previous word -- in which case it's fine as it is.